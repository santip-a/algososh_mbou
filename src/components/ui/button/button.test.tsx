// import renderer from "react-test-renderer";
import { render, screen, fireEvent } from '@testing-library/react';
import { Button } from './button';

const renderer = require('react-test-renderer');

describe("Рендер кнопки", () => {
  test('Рендер кнопки с текстом', () => {
    const tree = renderer.create(<Button text="test" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('Рендер кнопки без текста', () => {
    const tree = renderer.create(<Button />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('Рендер заблокированной кнопки', () => {
    const tree = renderer.create(<Button text="test" disabled />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('Кнопка с лоадером', () => {
    const tree = renderer.create(<Button isLoader />).toJSON()
    expect(tree).toMatchSnapshot();
  });

  test('Нажатие кнопки', () => {
    const onClick = jest.fn();
    render(<Button text="test" onClick={onClick} />);
    const button = screen.getByText("test");
    fireEvent.click(button);
    expect(onClick).toHaveBeenCalled();
  });
})