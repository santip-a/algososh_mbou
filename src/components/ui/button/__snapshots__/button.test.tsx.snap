// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Рендер кнопки Кнопка с лоадером 1`] = `
<button
  className="text text_type_button text_color_primary button undefined loader "
  disabled={true}
  type="button"
>
  <img
    alt="Загрузка."
    className="loader_icon"
    src="loader.svg"
  />
</button>
`;

exports[`Рендер кнопки Рендер заблокированной кнопки 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  disabled={true}
  type="button"
>
  <p
    className="text undefined"
  >
    test
  </p>
</button>
`;

exports[`Рендер кнопки Рендер кнопки без текста 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;

exports[`Рендер кнопки Рендер кнопки с текстом 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  >
    test
  </p>
</button>
`;
